core/src/main/java/io/questdb/griffin/engine/groupby/vect/GroupByNotKeyedVectorRecordCursorFactory.java

```diff
@@ -158,6 +169,7 @@ static int runWhatsLeft(
 
     @Override
     protected void _close() {
+        Misc.freeObjListAndKeepObjects(frameMemoryPools);
```

Comment:
I think ` frameAddressCache.clear();` should also be called here.

Reply:
The cache doesn't hold native memory, so we don't need to clear it when the factory is freed. If it would be `Closeable`, then the `frameAddressCache.close()` would be necessary here.

Also, we clear the cache on the cursor close here and in `*.groupby.vect.GroupByRecordCursorFactory`.

WDYT?

Reply:
I think it's worth cleaning to follow the code pattern unless there is a penalty.
My understanding of the pattern is that factory `_close()` should release everything related to the current run before the factory is pooled

Reply:
> My understanding of the pattern is that factory `_close()` should release everything related to the current run before the factory is pooled

No, `_close()` (and `close()`) is only called when we free the factory. After this, it can't be reused. When returning a factory to the pool (e.g. query cache) we only close its cursor, not the factory itself.

That's why clearing any garbage-collectable stuff in the factory's `close()` method is nothing more than redundant CPU cycles.

Reply:
Ok, good to know, I must confused it with something else